src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingProviderFactory.java

```diff
@@ -31,12 +31,14 @@ EmbeddingProvider create(
         EmbeddingProviderConfigStore.RequestProperties requestProperties,
         String baseUrl,
         String modelName,
+        int dimension,
         Map<String, Object> vectorizeServiceParameter);
   }
 
   private static final Map<String, ProviderConstructor> providersMap =
       Map.ofEntries(
           Map.entry(ProviderConstants.OPENAI, OpenAiEmbeddingClient::new),
+          Map.entry(ProviderConstants.AZURE_OPENAI, AzureOpenAiEmbeddingClient::new),
```

Comment:
Minor nitpick: maybe rename `AzureOpenAiEmbeddingClient` -> `AzureOpenAIEmbeddingClient` for consistency with `VertexAIEmbeddingClient` below?

Reply:
Tough call: I follow `OpenAiEmbeddingClient` here... :-/

Reply:
@maheshrajamani @Hazel-Datastax I can change names of both if you prefer "AI" capitalization, but wanted to check before doing that